nativelib/src/main/scala/scala/scalanative/runtime/NativeThread.scala

```diff
@@ -94,7 +94,7 @@ object NativeThread {
     fromRawPtr[scala.Byte](castObjectToRawPtr(thread))
 
   object Registry {
-    private val _aliveThreads = new ConcurrentHashMap[Long, NativeThread]()
+    private val _aliveThreads = TrieMap.empty[Long, NativeThread]
```

Comment:
Is there still a plan to port `ConcurrentHashMap` at a later point? I'd have to read the paper again but I think it makes different performance trade-offs than `ConcurrentHashMap` to support some special features.

Reply:
Yes, at some point I wish we port all the collections from JSR-166. The implementation ported from Scala.js is great, but I'm worried that it's not safe enough for usage in a multithreading environment.

Reply:
> The implementation ported from Scala.js is great, but I'm worried that it's not safe enough for usage in a multithreading environment.

Oh, definitely cannot use the current implementation in multi-threading :) we rely a lot on `ConcurrentHashMap` in Cats Effect.